### PR TITLE
Assign roles to users

### PR DESCRIPTION
**Pull request by Andycutter**
### PT link
https://www.pivotaltracker.com/story/show/159732583

### User story
```
As a system owner,
In order to differentiate between the Users,
I would like to be able to apply different roles to Users.
```
### Other
We need to have discussion about if this simple boolean column added to a user solution is good enough. The user will have columns of subscriber, author and editor that can be set to true or false with `current_user.update_attribute :author, true`. And if it is sufficient to use `if current_user.author?` as a check. Otherwise we could use the `gem 'pundit'` or another simple Active Record enum.

### CODE REVIEW

**Review by Siianko — APPROVED** [review]

File: db/migrate/20180819144937_add_editor_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddEditorToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :editor, :boolean, default: false
+  end
+end
```
Comment (tochman) [inline db/migrate/20180819144937_add_editor_to_users.rb:5]: All these booleans..... Why not use `enum` for `:role` ?

**Review by tochman — CHANGES_REQUESTED** [review]
I have a problem with the extensive use of booleans to keep track of roles. I also don't see any real value for the user in this feature.

File: db/migrate/20180819144937_add_editor_to_users.rb (hunk 1 above)
Comment (Andycutter) [reply db/migrate/20180819144937_add_editor_to_users.rb:5]: The reasoning with my limited experience what, that it seemed as the simplest solution for our MVP, but I could be wrong.
What is the advantage of enum? We get only one column and then numbers for our 4 roles, but are there other advantages in functionality?

Comment (tochman) [reply db/migrate/20180819144937_add_editor_to_users.rb:5]: There’s a guide about role based authorization in your course materials

**Pull request closed without merging**